index.js

```diff
@@ -720,30 +607,67 @@ TChannelConnection.prototype.send = function send(options, frame, callback) {
     //  throw new Error('duplicate frame id in flight');
     // }
 
-    frame.header.id = id;
-    frame.header.seq = 0;
+    var frame = self.handler.buildRequestFrame({
```

Comment:
Might be nicer as `sendCalllRequest()`. This will move the `_writeFrame` call into the `writeFrame` function of the handler.